File: src/main/java/com/github/graqr/threshr/model/products/ShippingOptions.java
package com.github.graqr.threshr.model.products;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record ShippingOptions(
        @JsonProperty("availability_status")
        String availabilityStatus,
        @JsonProperty("loyalty_availability_status")
        String loyaltyAvailabilityStatus,
        List<?> services) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/ShippingOptions.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class ShippingOptions(
    @param:JsonProperty("availability_status") val availabilityStatus: String,
    @param:JsonProperty("loyalty_availability_status") val loyaltyAvailabilityStatus: String,
    val services: List<Any>
) {}
